src/web/components/SharingPermission/BulkAddPermissions.tsx

```diff
@@ -42,70 +46,135 @@ export function BulkAddPermissions({
     },
   });
 
-  const { register, handleSubmit } = formMethods;
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isDirty },
+  } = formMethods;
+
+  const watchPublisherChecked = watch('publisherChecked');
+  const watchAdvertiserChecked = watch('advertiserChecked');
+  const watchDSPChecked = watch('DSPChecked');
+  const watchDataProviderChecked = watch('dataProviderChecked');
 
   const handleSave = (data: BulkAddPermissionsForm) => {
-    // TODO: call onBulkAddSharingPermission with the appropriate data
+    onBulkAddSharingPermission(getCheckedParticipantTypeIds(data, participantTypes));
   };
 
-  const onToggleShowRecommendedParticipants = () => {
+  const onToggleViewRecommendedParticipants = () => {
     setShowRecommendedParticipants((prevToggle) => !prevToggle);
-    // TODO: show the actual participants
+    // TODO: show/hide the actual participants
   };
 
+  const bulkAddPermissionsMessage = (
+    <>
+      <b>{getRecommendationMessageFromTypeNames(currentParticipantTypeNames)}</b>
+      <p>
+        Allow all current and future participants within a participation type to decrypt your UID2
+        tokens.
+      </p>
+    </>
+  );
+
+  const viewRecommendedParticipantsButton = (
+    <button
+      type='button'
+      className='text-button bulk-add-permissions-view-recommendations-button'
+      onClick={onToggleViewRecommendedParticipants}
+    >
+      {showRecommendedParticipants ? 'Hide Participants' : 'View Participants'}
+    </button>
+  );
+
+  const participantTypeCheckboxes = (
+    <div className='participant-type-checkbox-section'>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('publisherChecked')}
+      />
+      <span className='checkbox-label'>Publisher</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('advertiserChecked')}
+      />
+      <span className='checkbox-label'>Advertiser</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('DSPChecked')}
+      />
+      <span className='checkbox-label'>DSP</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('dataProviderChecked')}
+      />
+      <span className='checkbox-label'>Data Provider</span>
+    </div>
+  );
+
+  const saveRecommendationsButton = (
+    <div className='bulk-add-permissions-footer'>
+      <button type='submit' className='primary-button bulk-add-permissions-submit-button'>
+        Save Recommendations
+      </button>
+    </div>
+  );
+
+  const recommendationContent = (
+    <div className='bulk-add-permissions'>
+      <div className='bulk-add-permissions-body'>
+        {bulkAddPermissionsMessage}
```

Comment:
nit: I could see the common components used between recommendationContent and collapsibleContent. I wonder in this case if we change `content` props to become `children`
then we could do 
```
<Collapsible>
    <div className='bulk-add-permissions'>
      <div className='bulk-add-permissions-body'>
        {bulkAddPermissionsMessage}
        {participantTypeCheckboxes}
       {sharingParticipant && <Warning>}
{saveRecommendationsButton}
      </div>
</Collapsible>
```
is it easier to mantain?

Reply:
yeah it's a good question, I was wondering if there was a better way. I like the idea, though in principle it would be a bit more complicated, something like 
```
    <div className='bulk-add-permissions'>
      <div className='bulk-add-permissions-body'>
        {bulkAddPermissionsMessage}
        {participantTypeCheckboxes}
        {hasSharingParticipants &&
          hasUncheckedDefaultParticipantTypeRecommendation(
            currentParticipantTypeNames,
            watchPublisherChecked,
            watchAdvertiserChecked,
            watchDSPChecked,
            watchDataProviderChecked
          ) && (
            <div className='remove-recommended-type-warning'>
              <Banner
                type='warning'
                message='If you remove the sharing permissions for a participant type, all sharing permissions of that type are removed, including future participants of that type.'
              />
            </div>
          )}
        {viewRecommendedParticipantsButton}
      </div>
      {!hasSharingParticipants || (hasSharingParticipants && isDirty && saveRecommendationsButton)}
    </div>
```

Plus the need to alter the styling based on the two states and the `defaultOpen` prop. 

It's not clear to me which one would be better - what do you think?

Reply:
Ah, I see your point. Yep, passing with component is better this time. But could we still rename the `content` to `children` I think in that case it will easier to use later

Reply:
cool - so to clarify - leave the implementation basically as is, but pass it in as `children` instead of the `content` prop?

Reply:
That's right

Reply:
done!